Mark PatternAcceptanceTest as ported to the TCK

Mats-SX opened the pull request:
Should only be merged after https://github.com/opencypher/openCypher/pull/66 is merged.

petraselmer commented on community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/PatternExpressionAcceptanceTest.scala:
```diff
@@ -315,6 +336,7 @@ class PatternExpressionAcceptanceTest extends ExecutionEngineFunSuite with Match
     ))
   }
 
+  // TCK'd
```
I don't know if we should mark these tests - ones which include introspecting the plan, as per line 353 - as TCK'ed...that is, the actual test is TCK'ed, but not every aspect as envisaged by this test. I have come across similar. Maybe we ought to discuss...

Mats-SX replied:
Yes, perhaps these tests should receive a different annotation.

Mats-SX committed:
Minor cleanup

Mats-SX commented:
@petraselmer I've done this now in a bit of a different way. I removed all result assertions, and moved the tests with plan assertions to a differently named test class. The ported tests are removed, as per our new strategy.

@boggle I'd be happy for you to see this as well.

Mats-SX commented:
Deal with case expression.

Mats-SX committed:
Mark tests as ported to the TCK

The use of `CASE` expressions and `extract()` are kept, as they are not
to be included in the first cut of openCypher.

Implementation detail assertions (e.g. on plan operators) are kept.

Mats-SX commented:
@boggle This should be mergable now. Only acceptance tests with openCypher features are removed.
